src/Symfony/Component/String/Utf8String.php

```diff
@@ -0,0 +1,331 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\String;
+
+use Symfony\Component\String\Exception\ExceptionInterface;
+use Symfony\Component\String\Exception\InvalidArgumentException;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ * @author Hugo Hamon <hugohamon@neuf.fr>
+ *
+ * @throws ExceptionInterface
+ */
+class Utf8String extends AbstractString
```

Comment:
Wouldn't `UnicodeString` be a better name for this class? After all, at the code point level it doesn't really matter anymore how the string was encoded to binary. It could have been utf-8, it could have been utf-16, the string may have never been encoded to binary at all (directly constructed from code points); at code point level they are all the same. Of course you need to pick some encoding internally (since PHP's string are all byte strings), but the implementation should be able to hide that completely from the outside. One way to make this explicit would be to make the constructor private and offer a `fromUtf8` static constructor. That way a `fromUtf16` static constructor (which converts to utf8 for internal use) would be possible just as well.

Reply:
but both the code point and grapheme strings give you UTF-8. With your argument, both should be named `Utf8String`.
Btw, with many people knowing more `UTF-8` than `Grapheme`, I think the current name might push them to the wrong implementation if your expectation is that most case should use GraphemeString.